Widgets_CDAT/DictWidget/KeyValueEditor.py

```diff
@@ -0,0 +1,274 @@
+from PySide.QtCore import *
+from PySide.QtGui import *
+from functools import partial
+import pdb
+
+class InputChecker(QValidator):
+
+    inputInvalid = Signal()
+    correctInput = Signal()
+
+    def __init__(self, index):
+        super(InputChecker, self).__init__()
+        self.texts = []
+        self.index = index
+
+    def validate(self, input, pos):
+        if input in self.texts:
+            if self.texts.index(input) != self.index:
+                self.inputInvalid.emit()
+                return QValidator.Intermediate
+
+        self.correctInput.emit()
+        return QValidator.Acceptable
+
+    def setIndex(self, index):
+        self.index = index
+
+
+class DictEditor(QWidget):
+    dictEdited = Signal(dict)
+
+    def __init__(self):
+        super(DictEditor, self).__init__()
+        self.valid_keys = None
+        self.keys = []
+        self.values = []
+        self.textEdits = []
+        self.valueEdits = []
+        self.validators = []
+        self.row_count = 0
+        self.rows = QVBoxLayout()
+        self.button = QVBoxLayout()
```

Comment:
You don't need to keep a reference to this object; Qt should handle it all for you